asapdiscovery-docking/asapdiscovery/docking/workflows/large_scale_docking.py

```diff
@@ -0,0 +1,359 @@
+    prepped_complexes = prepper.prep(
+        complexes, use_dask=inputs.use_dask, dask_client=dask_client
+    )
+    del complexes
+
+    if inputs.gen_du_cache:
+        logger.info(f"Generating DU cache at {inputs.gen_du_cache}")
+        prepper.cache(prepped_complexes, inputs.gen_du_cache)
+
+    # define selector and select pairs
+    # using dask here is too memory intensive as each worker needs a copy of all the complexes in memory
+    # which are quite large themselves, is only effective for large numbers of ligands and small numbers of complexes
+    logger.info("Selecting pairs for docking based on MCS")
+    selector = MCSSelector()
+    pairs = selector.select(
+        query_ligands,
+        prepped_complexes,
+        n_select=inputs.n_select,
+        use_dask=False,
+        dask_client=None,
+    )
+
+    del prepped_complexes
+
+    # dock pairs
+    logger.info("Running docking on selected pairs")
+    docker = POSITDocker()
+    results = docker.dock(
+        pairs,
+        use_dask=inputs.use_dask,
+        dask_client=dask_client,
+    )
+
+    del pairs
+
+    # write docking results
+    logger.info("Writing docking results")
+    POSITDocker.write_docking_files(results, output_dir / "docking_results")
+
+    # score results
+    logger.info("Scoring docking results")
+    scorer = MetaScorer(
+        scorers=[
+            ChemGauss4Scorer(),
+            GATScorer.from_latest_by_target(inputs.target),
+            SchnetScorer.from_latest_by_target(inputs.target),
+        ]
+    )
+
+    if inputs.write_final_sdf:
+        logger.info("Writing final docked poses to SDF file")
+        write_ligands_to_multi_sdf(
+            output_dir / "docking_results.sdf", [r.posed_ligand for r in results]
+        )
+
+    scores_df = scorer.score(
+        results, use_dask=inputs.use_dask, dask_client=dask_client, return_df=True
+    )
+
+    del results
+
+    scores_df.to_csv(data_intermediates / "docking_scores_raw.csv", index=False)
+
+    logger.info("Filtering docking results")
+    # filter for POSIT probability > 0.7
+    scores_df = scores_df[
+        scores_df[DockingResultCols.DOCKING_CONFIDENCE_POSIT.value]
+        > inputs.posit_confidence_cutoff
+    ]
+
+    check_empty_dataframe(
+        scores_df,
+        logger=logger,
+        fail="raise",
+        tag="scores",
+        message="No docking results passed the POSIT confidence cutoff",
+    )
+
+    # filter out clashes (chemgauss4 score > 0)
+    scores_df = scores_df[scores_df[DockingResultCols.DOCKING_SCORE_POSIT] <= 0]
+
+    check_empty_dataframe(
+        scores_df,
+        logger=logger,
+        fail="raise",
+        tag="scores",
+        message="No docking results passed the clash filter",
+    )
+
+    # then order by chemgauss4 score and remove duplicates by ligand id
+    scores_df = scores_df.sort_values(
+        DockingResultCols.DOCKING_SCORE_POSIT.value, ascending=True
+    )
```

Comment:
Looks like the ML scores are not actually used at all?

Reply:
ML scores pushed to postera but not used for ranking at the moment.